src/bootstrap/test.rs

```diff
@@ -1030,15 +1030,26 @@ impl Step for Compiletest {
                 if let Some(ar) = builder.ar(target) {
                     cmd.arg("--ar").arg(ar);
                 }
+
+                // Add the llvm/bin directory to PATH since it contains lots of
+                // useful, platform-independent tools
+                let llvm_bin_path = llvm_config.parent()
+                    .expect("Expected llvm-config to be contained in directory");
+                assert!(llvm_bin_path.is_dir());
+                let old_path = env::var_os("PATH").unwrap_or_default();
+                let new_path = env::join_paths(iter::once(llvm_bin_path.to_path_buf())
+                                               .chain(env::split_paths(&old_path)))
+                               .expect("");
+                cmd.env("PATH", new_path);
```

Comment:
I believe this may be the bit which overrides the `PATH` set by `cc` to include that extra directory to find `mspdb140.dll`.